File: ORB.Core/Models/CorporateRiskMitigationAction.cs
using System;
using System.Collections.Generic;

namespace ORB.Core.Models
{
    public partial class CorporateRiskMitigationAction : RiskMitigationAction
    {
        public CorporateRiskMitigationAction() : base()
        {
            CorporateRiskRiskMitigationActions = new HashSet<CorporateRiskRiskMitigationAction>();
            RiskMitigationActionUpdates = new HashSet<CorporateRiskMitigationActionUpdate>();
        }

        public virtual CorporateRisk Risk { get; set; }
        public virtual ICollection<CorporateRiskRiskMitigationAction> CorporateRiskRiskMitigationActions { get; set; }
        public virtual ICollection<CorporateRiskMitigationActionUpdate> RiskMitigationActionUpdates { get; set; }
    }
}

File: corporate-reporting/src/types/CorporateRiskMitigationAction.ts
import { ICorporateRiskRiskMitigationAction } from ".";
import { ICorporateRisk } from "./CorporateRisk";
import { IRiskMitigationAction, RiskMitigationAction } from "./RiskMitigationAction";
import { IRiskMitigationActionUpdate } from "./RiskMitigationActionUpdate";

export interface ICorporateRiskMitigationAction extends IRiskMitigationAction {
    Risk?: ICorporateRisk;
    CorporateRiskRiskMitigationActions?: ICorporateRiskRiskMitigationAction[];
    RiskMitigationActionUpdates?: IRiskMitigationActionUpdate[];
}

export class CorporateRiskMitigationAction extends RiskMitigationAction implements ICorporateRiskMitigationAction { 
    public CorporateRiskRiskMitigationActions = [];
}
